### PR TITLE
Helen/post tasks

### PR DESCRIPTION
**Pull request by HacksForHugs**
Created rake task for batch posting tasks on the day the tasks are due, as well as a method of storing and retrieving the status of the tasks once they've been completed

### CODE REVIEW

File: app/models/recurrence.rb [hunk 1]
```diff
@@ -18,4 +18,48 @@
 class Recurrence < ActiveRecord::Base
   belongs_to :pickup
   enum day: [:monday, :tuesday, :wednesday, :thursday, :friday]
+
+  def location
+    self.pickup.location
```
Comment (warrenshen) [inline app/models/recurrence.rb]: You don't need `self`. You can also do `belongs_to :location, through: :pickup` and then you don't have to define this method.

File: app/models/recurrence.rb [hunk 2]
```diff
@@ -18,4 +18,48 @@
...
+  end
+
+  def business
+    self.pickup.location.business
+  end
```
Comment (warrenshen) [inline app/models/recurrence.rb]: Should change once you change above

File: app/models/recurrence.rb [hunk 3]
```diff
@@ -18,4 +18,48 @@
...
+      else
+        resp = OnfleetAPI.post_task(r)
+      end
+      if resp.key?('id')
+        task_id = resp["id"]
+        r.update(task_id: task_id)
+      else
+        f << {:failed => r.id, :message => resp["message"]}
+      end
+      sleep 0.2
```
Comment (warrenshen) [inline app/models/recurrence.rb]: This hardcoded sleep seems kind of dangerous

File: db/migrate/20161119232342_add_driver_torecurrence.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class AddDriverTorecurrence < ActiveRecord::Migration
+  def change
+    add_column :recurrences, :driver_id, :string, null: false, default: ''
+    add_column :recurrences, :task_id, :string
```
Comment (warrenshen) [inline db/migrate/20161119232342_add_driver_torecurrence.rb]: Are these foreign keys - if so, shouldn't they be `ints`?

File: app/models/recurrence.rb (hunk 3 above)
Comment (warrenshen) [inline app/models/recurrence.rb]: Makes sense - just leave a comment explaining why `0.2`!

**Comment by eric-ls** [comment]
👏

**Pull request merged**